Update 40x46mm.json

MichaelMacha opened the pull request:
#### Summary
Content "Added a .22 LR Hornet's Nest round and a .410 Hornet's Nest round"

#### Purpose of change

.22 LR hornet's nest rounds were suggested on Reddit, as a use for .22 LR rounds in general. They are typically very underpowered against mid-game enemies; but the hornet's nest combines ten of them into an adapter and allows them all to be fired at once from a 40mm launcher.

.410 hornet's nest rounds are a similar concept, using the less-useless .410 shot round. These also fire all four rounds at once.

#### Describe the solution

I simply added hornet's nest rounds to the JSON. I have no yet added crafting recipes or disassembly (which should yield at least ten .22 LR rounds), as of the moment; and anyone's welcome to do as much.

ADDENDUM: I also added the .410 shot hornet's nest round.

#### Describe alternatives you've considered

No reason to consider any alternatives.

#### Testing

The change is small. I ran it on version 11406, and it worked out well.

ADDENDUM: I updated my last version to use ten 22lr rounds instead of six, and the .410 hornet's nest; both work well on 11406 on my machine.

#### Additional context

.22 LRs are generally meant to be discharged from a "long rifle", not a stocky grenade launcher. I increased the dispersion substantially in the JSON to account for this, but if someone knows the numbers better than me it could be reviewed.

It would be good to have the recipe for these items in an advanced bullet or gunsmithing related book, which is probably already in the game; but I haven't gotten around to it. It would definitely add value to the .22 LRs we run into, for anything other than hunting small game. All I know is that they are generally made out of aluminum for the sake of managing round weight, have a little additional primer in there (pistol primer?), and in this case ten .22 LRs. (Or, for the .410, four .410 shot shells.)

It might also be neat if they had to be loaded; but I can't think of a situation in which they would be unloaded other than by firing them. It might be easier to make them fixed objects.

MichaelMacha commented:
Christ, I just realized these things are supposed to fire _ten_ .22 LRs, not six. Sorry guys I've had a mild headache all day. It should be easy enough to fix, right?

Maleclypse commented:
The below will fix your summary.

`Summary
Content "Added a .22 LR Hornet's Nest round"`

Shodan14 commented:
Wow, amazing to see my meme cross-post become a reality in less than 24h. Love the community enthusiasm!

MichaelMacha commented:
> I guess the "grenade" bit should be reusable?

Probably. I'm thinking that might get a little complicated; the graphics I've looked at clearly show that the .22 LR (or whatever it is) can be loaded in by hand. However, I've been treating this like a shell. I'd have to look into how to make a shell reloadable; and moreover, we would need a very specific message as to how it's used, so people don't get confused.

Moreover, I would have to determine what happens when you only use a _partially_ reloaded shell—like four .22 LRs instead of the full six—and at that point it's a bit more than I'm capable of spending time on right now.

If somebody knows how to handle that it would be great; but at the moment I'm still a little buried on my other project and am simply not familiar enough with the JSON design. If somebody knows how to do this it would be great.

MichaelMacha commented:
I went ahead and added the .410 shot hornet's nest shells, and updated the 22lr to use ten rounds instead of six. I also discovered that neat little "dispersion" tag and _think_ I've set it to something appropriate; but I'm not that familiar with the units. In-game it's been quite usable.

Shodan14 commented:
[Here's a video of someone loading and using one for reference](https://www.youtube.com/watch?v=bGXvMb7Q57Y).

casswedson commented on data/json/items/ammo/40x46mm.json:
```diff
@@ -179,5 +179,33 @@
+    "type": "AMMO",
+    "name": { "str_sp": "40x46mm .22 LR hornet's nest" },
+    "proportional": { "dispersion": 1.8 },
+    "description": "A set of ten .22 LR bullets compressed into a 40mm adapter. Fires all ten rounds simultaneously.",
```
```suggestion
    "description": "A set of ten .22 LR bullets compressed into a 40mm adapter.  Fires all ten rounds simultaneously.",
```

casswedson commented on data/json/items/ammo/40x46mm.json:
```diff
@@ -179,5 +179,33 @@
+    "type": "AMMO",
+    "name": { "str_sp": "40x46mm .410 hornet's nest" },
+    "proportional": { "dispersion": 1.4 },
+    "description": "A set of four .410 shells compressed into a 40mm adapter. Fires all four rounds at once.",
```
```suggestion
    "description": "A set of four .410 shells compressed into a 40mm adapter.  Fires all four rounds at once.",
```

MichaelMacha commented:
> [Here's a video of someone loading and using one for reference](https://www.youtube.com/watch?v=bGXvMb7Q57Y).

Interesting, that grouping was a lot tighter than I thought it would be. Also, those bullets were facing pretty much every single direction other than the one they were traveling in—but hey, it's .22 _LR_ for a reason.

This also definitely looks reusable; it seems you can reload the shell. I didn't see any kind of primer in there other than that in the bullets; so perhaps that isn't necessary--though notably, this was an 18-round 40mm shell and not the one I was modeling off of. It seems more like an adapter for the hammer.

Shodan14 commented:
> 
> 
> > [Here's a video of someone loading and using one for reference](https://www.youtube.com/watch?v=bGXvMb7Q57Y).
> 
> Interesting, that grouping was a lot tighter than I thought it would be. Also, those bullets were facing pretty much every single direction other than the one they were traveling in—but hey, it's .22 _LR_ for a reason.
> 
> This also definitely looks reusable; it seems you can reload the shell. I didn't see any kind of primer in there other than that in the bullets; so perhaps that isn't necessary--though notably, this was an 18-round 40mm shell and not the one I was modeling off of. It seems more like an adapter for the hammer.

Yeah, I don't think it has its own primer, it just hits all the .22LR ones at the same time. Probably not ideal, but seems like something that would be handy in a zombie survival situation.
